### DISCUSSION TITLE
in f5-tts_finetune-gradio how correct IndexError: too many indices for tensor of dimension 2

### DISCUSSION THREAD

**Discussion by paulokanda** [SWivid/F5-TTS#679] [General]
Hi,
any Idea about this error in test model?

in f5-tts_finetune-gradio how correct IndexError: too many indices for tensor of dimension 2

thanks in advance

**Comment by ephr0n** [comment]
Having the same issue, did you ever find a fix?

**Comment by Valoria191** [comment]
same issue, if somone know how to fix it please

**Comment by Valoria191** [comment]
I couldn't solve the problem, but I can suggest an alternative solution with python:
(found with chatGPT, work for me)


import torch
from tortoise.api import TextToSpeech
import torchaudio


model_path = "C:/Users/......../....../F5-TTS/ckpts/your_model/model_last.pt"


tts = TextToSpeech()


custom_state = torch.load(model_path, map_location="cpu")


tts.autoregressive.load_state_dict(custom_state, strict=False)


text = "Hello, I'm a personalised text-to-speech model."


audio = tts.tts_with_preset(
    text,
    preset="fast",  # Choice : "fast", "standard", "high_quality"
)


audio = torch.tensor(audio).squeeze().unsqueeze(0)  # Supprime les dimensions inutiles et garde le bon format


output_path = "C:/Users/......../....../audio/audio.wav"
sample_rate = 24000
torchaudio.save(output_path, audio, sample_rate)

print(f"Voice synthesis saved in : {output_path}")

**Reply by kostianu4** [reply]
in F5-TTS\src\f5_tts\model\modules.py
after this:
```
        if rope is not None:
            freqs, xpos_scale = rope
```
Added this:
            
```
            if freqs.dim() == 2:

                freqs = freqs.unsqueeze(0)
```
                
Chatgpt advice
Error now dissapeared, will test how it good or not
